test/plugin.test.js

```diff
@@ -0,0 +1,92 @@
+// -------------------------------------
+// jest + chai + sinon + sinon-chai
+let chai = require("chai");
+let sinon = require("sinon");
+let sinonChai = require("sinon-chai");
+let expect = chai.expect;
+  chai.use(sinonChai);
+// ------------------------------------
+let _ = require('lodash')
+
+let tdd = require('./_tdd')
+let serverless_express = require('serverless-express')
+let mock = require('./src/mock')
+
+let initPlugin = require('./src/init-plugin')
+
+describe('serverless-express plugin', function() {
+
+    it('should call initialize method when the plugin gets instanciated', function(){
+      let sls_plugin = require('serverless-express')
+      let spy_initilialize = sinon.spy(sls_plugin.prototype, '_initilialize' );
+
+      let params = mock.generate()
+      let serverless = params[0]
+      let options = params[1]
+      new sls_plugin(serverless, options)
+
+      // console.log( expect(spy_initilialize).to.have.been.called )
+      expect(spy_initilialize).to.have.been.called
+
+    })
+
+    it('should initialize the plugin in the right sequence', function(){
+      let sls_plugin = require('serverless-express')
+      let spy_testPlatform = sinon.spy(sls_plugin.prototype, 'testPlatform' );
+      let spy_setPlatformEnvironment = sinon.spy(sls_plugin.prototype, 'setPlatformEnvironment' );
+ 
+      let params = mock.generate()
+      let serverless = params[0]
+      let options = params[1]
+      new sls_plugin(serverless, options)
+
+      expect(spy_testPlatform).to.have.been.called
+      expect( spy_setPlatformEnvironment).to.have.been.calledAfter( spy_testPlatform )
+
+    })
+
+    it('should throw an error if the provider is not  supported yet', function(){
+      
+      let params = mock.generate()
+      let serverless = params[0]
+      let options = params[1]
+
+      expect(()=>{ 
+        serverless.variables.service.provider.name = 'fakeCloudProviderName'
+        new serverless_express(serverless, options) 
+      }).to.throw()
+
+    })
+
+    it('should not throw an error if provider is supported', function(){
+
+      let params = mock.generate()
+      let serverless = params[0]
+      let options = params[1]
+
+      expect(()=>{ 
+        tdd.supported_providers.forEach((providerName)=>{
+          serverless.variables.service.provider.name = providerName
+          new serverless_express(serverless, options) 
+        })
+      }).to.not.throw()
+
+    })
+
+
+    it("should set providerName as environment variable when it's a supported provider", function(){
```

Comment:
why does the plugin set an env var? Do other serverless plugins do this?

Reply:
```serverless-express/express``` and ```serverless-express/handler``` rely on provider/platform detection in order to run the express app correctly.

Setting ```SERVERLESS_EXPRESS_PLATFORM``` provides a unified and simple way of knowing in which platform are we running the function. 

Otherwise, we would have to implement this same functionality by checking ```process.env``` and infer the provider name with booleans like 
```const isAws = ( process.env.AWS_LAMBDA_FUNCTION_NAME? true : false )```.

The thing is that I couldn't find any official documentation about those default environment variables for Aws Lambda nor Google Cloud functions and therefore I don't think we should rely on the second option since there is no official documentation about this behavior.

What do you think? 
Maybe it should be mentioned in the readme.md?

Reply:
I think as long as it is mentioned in the README it is ok. I just don't like the idea of a plugin mutating the environment, there is an (unlikely) chance that a client might be relying on that same variable name.

Reply:
the variable name is namespaced ```SERVERLESS_EXPRESS_PLATFORM```.
like you said, I will mention it in the readme.

Otherwise, do you see an alternative that? that maintain provider detection and code stability?

Reply:
I can't think of a better solution atm so I think its fine to ship. Thanks again for reviving this project!

Reply:
Thanks ;)